infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java

```diff
@@ -458,6 +465,19 @@ private Properties getInfraInputs(TestPlan testPlan, Script script) {
         }
         props.putAll(script.getInputParameters());
 
+        //Fetch infra-parms from db and include their sub-props also as properties to be passed.
+        InfrastructureParameterUOW infrastructureParameterUOW = new InfrastructureParameterUOW();
+        Properties infraParamsAsProps =  testPlan.getInfrastructureConfig().getParameters();
+        for (String key: infraParamsAsProps.stringPropertyNames()) {
+            try {
+                infrastructureParameterUOW
+                        .getInfrastructureParameter(infraParamsAsProps.getProperty(key)).get()
+                        .getProcessedSubInfrastructureParameters()
+                        .forEach(infraParam -> props.setProperty(infraParam.getType(), infraParam.getName()));
```

Comment:
IMHO, infrastructure_parameter table is an abstract detail, and the AWSProvider should not be aware of it. If we couple these two, then it'll be, 1. hard to replace it with something else (for ex. a json file!), 2. we'll have to duplicate above logic in each and every concrete implementations of InfrastructureProvider.
 
So, I think we should always work with the TestPlan object beyond the InfrastructureCombinationProvider class.

WDYT?

Reply:
+1